### ISSUE TITLE
Syncing forked repository

### ISSUE THREAD

**Issue by ojha-aditya** [ubsuny/CP1-24-HW1#87]
So I am a little confused with the sync feature. Syncing a branch on our forked repository updates it with the current state of the actual repository *while retaining our own edits*? Or do we have to commit an edit before we sync for it to be retained?

**Comment by avgagliardo** [comment]
### Basics of Forking and Syncing
When you **fork** a repository, you create a copy of the original repository (the “upstream repository”) under your own GitHub account. This is your forked repo-- Over time, the upstream repository might receive new commits, while you might also be making changes in your fork.

The **sync** feature helps keep your forked repository up-to-date with the upstream repository.

### Regarding your question
Syncing your forked repository with the upstream repository does not automatically merge changes into the branch where you’ve made your own edits. 

It works like this:
1. Fetch Changes from Upstream: When you sync, you are asking GitHub to fetch the latest commits from the upstream repository (in this case, the shared class repo) and incorporating them into your own forked repo. But these upstream changes are fetched into your repository without overwriting your changes (in your repo, not locally).

2. Commit Your Changes First: If you have made changes in your local repository or on a branch in your fork, you will need to commit those changes before syncing. This is what will ensure that your edits are saved. Uncommitted changes will not be synced, and if there are conflicts between your local edits and the upstream changes, Git will prompt you to resolve those conflicts.

3. Merge Changes: After syncing (fetching updates from the upstream), you will typically merge those changes into your working branch. During this process, if any of the updates conflict with your own changes, Git will ask you to resolve those conflicts manually.

### In short, the pattern is:
1. Commit your changes (so they are saved in the current branch).

2. Sync the branch: Pull the latest changes from the upstream repository into your fork.

3. Merge the updates into your working branch if needed.

**Comment by ojha-aditya** [comment]
Got it! Thank you very much. So if I created a PR with an older version of the branch and now wish to update the same, do I create a new PR after syncing?

**Comment by avgagliardo** [comment]
> Got it! Thank you very much. So if I created a PR with an older version of the branch and now wish to update the same, do I create a new PR after syncing?

If you've already made some commits and create a PR, then anything else you commit afterwards (to the repository from which you created the PR) will automatically be included with the original PR-- until it is merged.

When you commit the new changes, you can check the original PR and you will see those changes included in the PR's feed.

**Comment by ojha-aditya** [comment]
I understand now, thank for help. Closing this.
